pennylane/ops/qubit/matrix_ops.py

```diff
@@ -65,20 +69,28 @@ def __init__(self, *params, wires, do_queue=True):
         # of wires fits the dimensions of the matrix
         if not isinstance(self, ControlledQubitUnitary):
             U = params[0]
+            U_shape = qml.math.shape(U)
 
             dim = 2 ** len(wires)
 
-            if qml.math.shape(U) != (dim, dim):
+            if not (len(U_shape) in {2, 3} and U_shape[-2:] == (dim, dim)):
                 raise ValueError(
-                    f"Input unitary must be of shape {(dim, dim)} to act on {len(wires)} wires."
+                    f"Input unitary must be of shape {(dim, dim)} or ({dim, dim}, batch_size) "
```

Comment:
Will `{dim, dim}` (the second one) render correctly, since this is a implied tuple?

Also, should the latter be `(batch_size, dim, dim)`?